### ISSUE TITLE
How to use ansible checker?

### ISSUE THREAD

**Issue by xenithorb** [scrooloose/syntastic#1788]
The wiki is very sparse on ansible information and I need to know how to enable it properly. 

I have tried various things before writing this, such as: 

```
     let g:syntastic_ansible_ansible_lint_exec = '/usr/bin/ansible-lint'
     let g:syntastic_ansible_checkers = ['ansible_lint']
```

I have also looked at the ansible_lint.vim source code, but that wasn't very revealing either.

Is this going to require a modeline on all files to work?

How will it differentiate between YAML and YAML+Ansible?

With an ansible-playbook YAML oriented file open, I have this is my `SyntasticInfo` output:

```
Syntastic version: 3.7.0-144 (Vim 704, Linux)
Info for filetype: yaml
Global mode: active
Filetype yaml is active
The current file will be checked automatically
Available checkers: -
Currently enabled checkers: -
```

Thanks

**Comment by lcd047** [comment]
> The wiki is very sparse on ansible information and I need to know how to enable it properly.

Well, the main information is in the manual.  There isn't anything special about `ansible-lint` once you understand how syntastic is supposed to work, but yes, the manual is not the best written technical piece.  Suggestions for improvements to the docs are always welcome.

> I have tried various things before writing this, such as:

`g:syntastic_ansible_checkers` is needed.  You can leave out `g:syntastic_ansible_ansible_lint_exec` if `ansible-lint` is in your `$PATH`.  The remaining piece is that you need to set the `filetype` of your playbooks to `ansible`.  Syntastic should start working after that.

> Is this going to require a modeline on all files to work?

Normally Vim should detect the correct `filetype` automatically, or you'd use some plugin to do it for you when that doesn't work.  I'm not using Ansible, but a search for `vim ansible` finds [ansible-vim](https://github.com/pearofducks/ansible-vim) and [vim-ansible-yaml](https://github.com/chase/vim-ansible-yaml) that look like they might work.

That said, setting `filetype` in a `modeline` would work too.

> How will it differentiate between YAML and YAML+Ansible?

It won't, detecting `filetype`s is outside the scope of syntastic.  It's up to you to make sure the `filetype` is set correctly for the files you want to check.

**Comment by xenithorb** [comment]
I gathered some of this from the README, but this was the extra push I needed. I've been using the plugin for a while now and haven't needed any special vim "recognizer" plugin yet. Thanks 

I understand better now that Syntastic is not responsible for recognizing the filetype if it is "incorrect".

--- 

I was able to quickly install ansible-vim (the other one is unmaintained and the first project looks the best) by cloning it into my bundle folder (since I use pathogen for Syntastic anyway) - once I did that It started working right away. 

Thanks!
